thunderfish/powerspectrum.py

```diff
@@ -0,0 +1,89 @@
+import numpy as np
+import matplotlib.mlab as ml
+
+def calc_psd(data, samplerate, fresolution):
+    """
+    Calculates a Powerspecturm.
+
+    This function takes a data array, its samplerate and a frequencyresolution for the powerspectrum.
+    With this input it first calculates a nfft value and later a powerspectrum.
+
+    :param data:                (1-D array) data array you want to calculate a psd of.
+    :param samplerate:          (float) sampling rate of the data that you want to calculate a psd of.
+    :param fresolution:         (float) frequency resolution of the psd.
+    :return:                    (2-D array) contains the power and frequency calculated in the powerspectrum.
+    """
+
+    nfft = int(np.round(2 ** (np.floor(np.log(samplerate / fresolution) / np.log(2.0)) + 1.0)))
+    if nfft < 16:
+        nfft = 16
+    power, freqs = ml.psd(data, NFFT=nfft, noverlap=nfft / 2, Fs=samplerate, detrend=ml.detrend_mean)
+    return [power, freqs]
+
+def powerspectrumplot(power, freqs, ax):
```

Comment:
Is this only needed for the main script?
This is so trivial, I would remove it.